Mathlib/Data/List/Rotate.lean

```diff
@@ -584,109 +576,103 @@ theorem length_cyclicPermutations_of_ne_nil (l : List α) (h : l ≠ []) :
     length (cyclicPermutations l) = length l := by simp [cyclicPermutations_of_ne_nil _ h]
 #align list.length_cyclic_permutations_of_ne_nil List.length_cyclicPermutations_of_ne_nil
 
-set_option linter.deprecated false in
 @[simp]
-theorem nthLe_cyclicPermutations (l : List α) (n : ℕ) (hn : n < length (cyclicPermutations l)) :
-    nthLe (cyclicPermutations l) n hn = l.rotate n := by
-  obtain rfl | h := eq_or_ne l []
-  · simp
-  · rw [length_cyclicPermutations_of_ne_nil _ h] at hn
-    simp [dropLast_eq_take, cyclicPermutations_of_ne_nil _ h, nthLe_take',
-      rotate_eq_drop_append_take hn.le]
-#align list.nth_le_cyclic_permutations List.nthLe_cyclicPermutations
+theorem cyclicPermutations_ne_nil : ∀ l : List α, cyclicPermutations l ≠ []
```

Comment:
Can `l : List α` be moved left of the colon?

Reply:
Why?

Reply:
If you want to move it, then please push to this branch. Otherwise, just merge it, please.

bors d=grunweg

Reply:
That is a fair point. (I was trying to understand why "hypothesis left of colon" was not followed - sounds like you have a good reason. Thanks for educating me!)